### PR TITLE
provider change detection change; Helm provider tag change

### PR DESCRIPTION
**Pull request by yanjiaxin534**

### CODE REVIEW

File: api/pkg/apis/v1alpha1/providers/target/helm/helm.go [hunk 1]
```diff
@@ -315,17 +315,24 @@ func (i *HelmTargetProvider) Get(ctx context.Context, deployment model.Deploymen
...
 				}
-
 				ret = append(ret, model.ComponentSpec{
 					Name: res.Name,
 					Type: "helm.v3",
 					Properties: map[string]interface{}{
 						"chart": map[string]string{
-							"repo":    repo,
-							"version": res.Chart.Metadata.Version,
+							"repo": repo,
```
Comment (msftcoderdjw) [inline api/pkg/apis/v1alpha1/providers/target/helm/helm.go:334]: Seems we don't carry over in version. So, for version upgrade, we will always run helm upgrade, is this right?

Comment (Haishi2016) [reply api/pkg/apis/v1alpha1/providers/target/helm/helm.go]: I think this is the result of changes getting crossed. We use a special tag "SYM:" to remember the chart repo, as I didn't know how to get it back with Helm SDK:

chart.Metadata.Tags = "SYM:" + helmProp.Chart.Repo //this is not used by Helm SDK, we use this to carry repo info

This was later updated to capture chart name as well:

chart.Metadata.Tags = "SYM-REPO:" + helmProp.Chart.Repo + ";SYM-NAME:" + helmProp.Chart.Name

Then, in the Get() method, we are trying to reconstruct repo and name from the tags, as seen in the code. I don't remember why the version is skipped - maybe a merging error on the way.

So, my proposal is:

1) Don't need to capture chart name in the special tag as we can read it back from chart metadata

chart.Metadata.Tags = "SYM:" + helmProp.Chart.Repo

Or, we can skip using the tag if there's a SDK native way to remember the repo

2) Play back the version during Get()

repo := ""
if strings.HasPrefix(res.Chart.Metadata.Tags, "SYM:") { //we use this special metadata tag to remember the chart URL
    repo = res.Chart.Metadata.Tags[4:]

...

"chart": map[string]string{
   "repo":    repo,
   "version": res.Chart.Metadata.Version,
   "name": res.Chart.Metadata.Name,
...

Please also add test cases to Get() after Apply() and verify the information is played back correctly.

Comment (yanjiaxin534) [reply api/pkg/apis/v1alpha1/providers/target/helm/helm.go]: I checked it. Get will return version name and repo. and other properties
image

File: docs/samples/canary/solution.yaml [hunk 2]
```diff
@@ -41,6 +41,7 @@ spec:
       deployment.replicas: "#1"
       container.ports: "[{\"containerPort\":5000,\"protocol\":\"TCP\"}]"
       container.image: "ghcr.io/eclipse-symphony/sample-flask-app:latest"
+      env.FOO: "bar"
```
Comment (msftcoderdjw) [inline docs/samples/canary/solution.yaml:44]: Seems this is a redundant change.

**Review by msftcoderdjw — CHANGES_REQUESTED** [review]

**Review by msftcoderdjw — APPROVED** [review]

**Pull request merged**